### PR TITLE
Add config management

### PR DESCRIPTION
**Pull request by StephaneTriomphe**
at init, and on stach status update, sink config internal cache is updated.
cached config values are used to generate proto buf messages.

### CODE REVIEW

File: lib/api/wpc_proto.h [hunk 1]
```diff
@@ -29,12 +29,29 @@ typedef enum
...
+ *               Pointer to gateway id string
+ * \param[in]    gateway_model
+ *               Pointer to gateway model string, "" if not available
+ * \param[in]    gateway_version
+ *               Pointer to gateway version string, "" if not available
+ * \param[in]    sink_id
+ *               Pointer to the sink id string
+ * \return       Return code of the operation
+ */
+app_proto_res_e WPC_Proto_initialize(char * gateway_id,
```
Comment (GwendalRaoul) [inline lib/api/wpc_proto.h:44]: For me, port_name and bitrate must remain in API.
WPC_Proto is wrapper on top of WPC_ so it looks a bit strange if you have to explicitly call WPC_initialize before  WPC_Proto_initialize

File: lib/wpc_proto/config.c [hunk 2]
```diff
@@ -0,0 +1,682 @@
+/* Copyright 2019 Wirepas Ltd licensed under Apache License, Version 2.0
```
Comment (GwendalRaoul) [inline lib/wpc_proto/config.c:1]: Please update date!

Comment (GwendalRaoul) [reply lib/wpc_proto/config.c:1]: Or use the same Copyright as other file from this repo

File: lib/wpc_proto/config.c [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+    uint16_t ac_limit_max;
+    uint16_t app_config_max_size;
+    uint16_t version[4];
+    uint8_t  max_mtu;
+    uint8_t  ch_range_min;
+    uint8_t  ch_range_max;
+    uint8_t  pdu_buffer_size;
+
+    /* Read parameters with node interrogation */
+    //uint16_t CurrentAC;
```
Comment (GwendalRaoul) [inline lib/wpc_proto/config.c:45]: Remove also commented field here

File: lib/wpc_proto/config.c [hunk 4]
```diff
@@ -0,0 +1,682 @@
...
+
+    res = (key_set_function)((uint8_t *) key);
+    return res;
+}
+
+/**
+ * \brief   Set cipher key handler
+ * \param   ... (from sd_bus property handler)
+ */
+static int set_cipher_key(sd_bus * bus,
```
Comment (GwendalRaoul) [inline lib/wpc_proto/config.c:105]: Please remove all this code, it has reference to dbus

File: lib/wpc_proto/config.h [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ */
+void Config_Fill_event_header(wp_EventHeader * header_p);
+
+/**
+ * \brief   Fill reponse message header
+ * \param   header_p pointer to header to fill
+ * \param   req_id request id
+ * \param   res response to request
+ */
+void Config_Fill_response_header(wp_ResponseHeader * header_p,
```
Comment (GwendalRaoul) [inline lib/wpc_proto/config.h:46]: Does it belong to config.h?

Shouldn't it be in a more generic module that could be used by other kind of messages, like data response?

**Review by GwendalRaoul — CHANGES_REQUESTED** [review]

File: lib/wpc_proto/config.c (hunk 4 above)
Comment (StephaneTriomphe) [reply lib/wpc_proto/config.c:107]: Done

File: lib/wpc_proto/config.c (hunk 3 above)
Comment (StephaneTriomphe) [reply lib/wpc_proto/config.c:46]: done

File: lib/wpc_proto/config.c (hunk 2 above)
Comment (StephaneTriomphe) [reply lib/wpc_proto/config.c:1]: done

File: lib/api/wpc_proto.h (hunk 1 above)
Comment (StephaneTriomphe) [reply lib/api/wpc_proto.h:44]: Done, will push after test

File: example/linux/proto-api/main.c [hunk 6]
```diff
@@ -56,14 +97,119 @@ int main(int argc, char * argv[])
...
+    sleep(2);
+    if (res_app != APP_RES_OK && res_app != APP_RES_STACK_ALREADY_STOPPED)
+    {
+        LOGE("Cannot stop stack %d\n", res_app);
+        goto exit_on_error;
+    }
+
+    LOGI("Stack is stopped\n");
+
+    // // Configure the node
```
[the remark this answers is not in the document]
Comment (StephaneTriomphe) [reply example/linux/proto-api/main.c:150]: Done

File: lib/wpc_proto/config.h (hunk 5 above)
Comment (StephaneTriomphe) [reply lib/wpc_proto/config.h:108]: this one (Config_Fill_response_header ) is used by both handle_send_data_request() and Config_Handle_set_config_request(). So it's generic. To be discuss where to put it.

**Pull request closed without merging**